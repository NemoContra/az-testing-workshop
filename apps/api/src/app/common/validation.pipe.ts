import { BadRequestException, Injectable, PipeTransform } from '@nestjs/common';
import { createFlightSchema } from '@az-testing-workshop/shared/util/api-models';
import { ZodSchema } from 'zod';

@Injectable()
export class ValidationPipe implements PipeTransform {
  constructor(private readonly schema: ZodSchema) {
  }
  transform(value: unknown): boolean {
    try {
      return this.schema.parse(value);
    } catch (error) {
      throw new BadRequestException(error);
    }
  }
}
