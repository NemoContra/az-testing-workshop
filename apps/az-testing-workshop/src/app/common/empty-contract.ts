import { Contract } from '@az-testing-workshop/shared/util/api-models';

export const emptyContract: Contract = {
  contractNumber: '',
  id: '',
  start: '',
  end: '',
  premium: 0,
  person: {
    firstname: '',
    lastname: '',
    dateOfBirth: '',
  },
};
