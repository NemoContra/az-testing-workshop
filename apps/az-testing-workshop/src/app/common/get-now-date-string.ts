const fillString = '0' as const;
const getYear = (now: Date) => `${now.getFullYear()}`;
const getMonth = (now: Date) =>
  (now.getMonth() + 1).toString().padStart(2, fillString);
const getDay = (now: Date) => now.getDate().toString().padStart(2, fillString);

export const getNowDateString = () => {
  const now = new Date();
  return [getYear(now), getMonth(now), getDay(now)].join('-');
};
