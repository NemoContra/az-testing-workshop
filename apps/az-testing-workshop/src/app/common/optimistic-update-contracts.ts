import { Contract } from '@az-testing-workshop/shared/util/api-models';

export const optimisticUpdateContracts = (
  contracts: Contract[] | undefined,
  updatedContract: Contract
) =>
  contracts?.map((contract) =>
    contract.id === updatedContract.id ? updatedContract : contract
  );
