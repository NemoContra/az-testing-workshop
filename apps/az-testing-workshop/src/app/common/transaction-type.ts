export const transactionTypes = ['AenderungNachname', 'Kuendigung'] as const;
export type TransactionsType = (typeof transactionTypes)[number];

export const transactionQueryParamKey = 'transaktion';
export const transactionQueryParam = <T extends TransactionsType>(
  transactionType: T
) =>
  ({
    [transactionQueryParamKey]: transactionType,
  } as const);
