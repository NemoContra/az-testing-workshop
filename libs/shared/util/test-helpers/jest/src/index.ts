export * from './lib/create-spy-observer';
