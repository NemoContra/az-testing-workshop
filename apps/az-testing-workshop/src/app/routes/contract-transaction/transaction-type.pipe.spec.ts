import { createPipeFactory } from '@ngneat/spectator/jest';
import { TransactionTypePipe } from './transaction-type.pipe';
import { TransactionsType } from '../../common/transaction-type';

describe('TransactionTypePipe', () => {
  const createPipe = createPipeFactory(TransactionTypePipe);

  /**
   * We can test pipes with `createPipeFactory` from spectator
   */
  describe('with spectator', () => {
    it('should render the transactionType for "AenderungNachname"', () => {
      const spectator = createPipe(
        `<span>{{ 'AenderungNachname' | transactionType }}</span>`
      );
    });

    it('should render the transactionType for "Kuendigung"', () => {
      const spectator = createPipe(
        `<span>{{ 'Kuendigung' | transactionType }}</span>`
      );
    });
  });

  /**
   * We can test pipes by creating an instance manually and calling the transform method
   */
  describe('with constructor usage', () => {
    it.each`
      transactionType                                   | result
      ${'AenderungNachname' satisfies TransactionsType} | ${'Änderung des Nachnamen'}
      ${'Kuendigung' satisfies TransactionsType}        | ${'Vertragskündigung'}
    `(
      'should transform the transactionType for $transactionType',
      ({ transactionType, result }) => {
        const pipe = new TransactionTypePipe();
      }
    );
  });
});
