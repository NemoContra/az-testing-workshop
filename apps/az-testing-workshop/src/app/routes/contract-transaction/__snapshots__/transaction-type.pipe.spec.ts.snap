// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionTypePipe should render the transactionType for "AenderungNachname" 1`] = `
<lib-ngneat-host-component>
  <span>
    Änderung des Nachnamen
  </span>
</lib-ngneat-host-component>
`;

exports[`TransactionTypePipe should render the transactionType for "Kuendigung" 1`] = `
<lib-ngneat-host-component>
  <span>
    Vertragskündigung
  </span>
</lib-ngneat-host-component>
`;
